corehq/apps/domain/views/base.py

```diff
@@ -49,8 +57,8 @@ def select(request, domain_select_template='domain/select.html', do_not_redirect
                 or domain_obj.is_snapshot
             ):
                 try:
-                    from corehq.apps.dashboard.views import dashboard_default
-                    return dashboard_default(request, last_visited_domain)
+                    return HttpResponseRedirect(reverse(next_view or 'dashboard_default',
+                                                args=[last_visited_domain]))
```

Comment:
[likely to be deferred to a followup PR]
`next_view` defaults to `domain_homepage` above, but to `dashboard_default` here.  Why is that?  It'd be easier to follow what's happening if when `next_view` is initialized, a single fallback is provided right there.

Also, `dashboard_default` is itself a redirect view, which I assume is why it was returned rather than redirected to.
https://github.com/dimagi/commcare-hq/blob/6ad0fe798fe56a5aa177283409edeb25a5e39882/corehq/apps/dashboard/views.py#L42-L45
Guessing that view made sense at some point where there was more than one option for a default, but at this point I think it'd make sense to replace all usages with a redirect to `DomainDashboardView.urlname` or `'dashboard_domain'`.  I see only three usages of it, which are all modified here, so this seems like an appropriate time.

Reply:
I'm not inclined to fiddle with `domain_homepage` vs `dashboard_default`. `domain_homepage` has a bunch of checks - authentication, web vs mobile worker, etc - and can potentially send you back to the domain select page, whereas `dashboard_default` is (as you point out) just a single view. I agree this could be improved but don't want to get into it right now. I do agree it makes sense to remove `dashboard_default`, will do that in a followup.